### PR TITLE
[#1498] email for subscribers

### LINKED ISSUE #1498: Add email notification for interested
**Is your feature request related to a problem? Please describe.**
Now we don't have any notification for users to inform them about newly added services or changes in services. 
**Describe the solution you'd like**
We should add notification in two cases:
1. New service in Category of interest 
2. New service in Research area of interest  

**Additional context**
@MinLidia will  provide notification templates after external consulting

### PR DESCRIPTION
**Pull request by goreck888**
Add email notification for subscribers
in categories and research areas
Fixes #1498

### CODE REVIEW

**Comment by agpul** [comment]
notify me if we have new templates inside

File: app/mailers/service_mailer.rb [hunk 1]
```diff
@@ -11,4 +11,27 @@ def new_question(recipient_email, author, email, text, service)
          subject: "Question about #{@service.title} service",
          template_name: "new_question")
   end
+
+  def new_service(service, common_categories, common_research_areas, subscriber_email)
+    @service = service
+    @common_categories = common_categories
+    @common_research_areas = common_research_areas
+    if @service.logo.attached?
+      puts "DUPA"
```
Comment (martaswiatkowska) [inline app/mailers/service_mailer.rb:20]: seriously?

File: app/mailers/service_mailer.rb [hunk 2]
```diff
@@ -11,4 +11,27 @@ def new_question(recipient_email, author, email, text, service)
...
+      if ActiveStorage::Blob.service.respond_to?(:path_for)
+        puts "CHUJ"
```
Comment (martaswiatkowska) [inline app/mailers/service_mailer.rb:22]: ...

File: app/mailers/service_mailer.rb [hunk 3]
```diff
@@ -11,4 +11,27 @@ def new_question(recipient_email, author, email, text, service)
...
+        attachments.inline["logo.png"] = File.read(ActiveStorage::Blob.service.send(:path_for, @service.logo.key))
+      elsif ActiveStorage::Blob.service.respond_to?(:download)
+        puts "CYCE"
```
Comment (martaswiatkowska) [inline app/mailers/service_mailer.rb:25]: ...

File: spec/mailers/service_spec.rb [hunk 4]
```diff
@@ -29,4 +29,16 @@
       expect(encoded_body).to have_content("text message")
     end
   end
+
+  context "publish" do
+    let(:user) { create(:user_with_interests) }
+    let(:service1) { create(:service, research_areas: user.research_areas) }
+    let(:service2) { create(:service, categories: user.categories) }
+    let(:mail0) { described_class.new_service(service1, user.categories, user.research_areas, user.email).deliver_now }
```
Comment (martaswiatkowska) [inline spec/mailers/service_spec.rb:37]: what mean mail0?

Comment (goreck888) [reply spec/mailers/service_spec.rb:37]: I had to use different name than `mail` because it's restricted for `ActionMailer::Base` method.

Comment (martaswiatkowska) [reply spec/mailers/service_spec.rb:37]: imo it's better to not use numbers. Maybe new_service_mail

**Pull request merged**